ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml

```diff
@@ -99,23 +130,40 @@ QtObject {
             }
         }
 
-        var amountStr
+        var amountStr = ""
+        var amountSuffix = ""
+
         let minAmount = 10**-optDisplayDecimals
-        if (currencyAmount.amount > 0 && currencyAmount.amount < minAmount && !optShowOnlyAmount)
+        if (currencyAmount.amount > 0 && currencyAmount.amount < minAmount && !optRawAmount)
         {
             // Handle amounts smaller than resolution
-            amountStr = "<%1".arg(numberToLocaleString(minAmount, optDisplayDecimals, locale))
+            amountStr = "<%1".arg(numberToLocaleString(minAmount, displayDecimals, locale))
         } else {
-            // Normal formatting
-            amountStr = numberToLocaleString(currencyAmount.amount, optDisplayDecimals, locale)
+            var amount
+            var displayDecimals
+            const numIntegerDigits = integralPartLength(currencyAmount.amount)
+            const maxDigits = 10 // 10 
+            // For large numbers, we use the short scale system and 2 decimal digits
+            // Handle large numbers
+            if (numIntegerDigits > maxDigits && !optRawAmount) {
+                amount = currencyAmount.amount/10**9
+                displayDecimals = 2
+                amountSuffix = "B"
```

Comment:
What does this "B" mean?

If it's visible to users... you know the drill :)

Reply:
added some comments in code and translation call, hope that was the whole drill

Reply:
Yeah thanks, that was it :)

But now we have an even more interesting discussion... but I guess "milliard" is no longer used in British English? @John-44 (billion AE vs. milliard BE)

Reply:
@caybro I've never heard of the word "milliard" before and I wasn't sure what it meant before I googled it, but @shamisem  from the Documentation team is probably the best person for language questions ;-)

I am of course aware of the difference between a 'US billion' (which = 1,000,000,000) and a 'UK billion' (which = 1,000,000,000,000 e.g. billion is a million millions), but a quick google led me to this House of Commons Library report (see https://researchbriefings.files.parliament.uk/documents/SN04440/SN04440.pdf ) which says that in 1974 the British prime minister of the time stated that he would be using the US definition of the word "billion" because the US meaning of the word had achieved greater international adoption.

The same report linked above also mentions a French connection to this confusion:

> billion, purposely formed in 16th c. to denote the second power of a million
(by substituting BI- prefix for the initial letters), trillion and quadrillion being
similarly formed to denote its 3rd and 4th powers. The name appears not to
have been adopted in Eng. before the end of the 17th … Subsequently the
application of the word was changed by French arithmeticians, figures being
divided in numeration into groups of threes, instead of sixes, so that F. billion,
trillion, denoted not the second and third powers of a million, but a thousand
millions and a thousand thousand millions. In the 19th century, the U.S.
adopted the French convention, but Britain retained the original and
etymological use (to which France reverted in 1948). 

There are a bunch of denomination descriptions where the UK and the US still differ, for example a UK "ton" is 1,016kg, while a US "ton" is 907kg.  Which is all very silly when there is the metric "ton" (invented in France) which is sensibly 1000kg.

So in the 3-way debate regarding what number the word "ton" actually referrers to, I have to say I agree with the French.

I should probably get back to work now! ;-)

Reply:
lol, I guess I'm too old then :) At school we were taught that in British English, the word being used is "milliard" (as opposed to AE, where the correct word would be "billion"). Btw, we still use the same as well in Czech (and I believe many other languages do as well)

https://www.merriam-webster.com/dictionary/milliard

Reply:
I'm all for naming them gigaDollars